src/components/TextField/TextField.vue

```diff
@@ -0,0 +1,253 @@
+<!--
+  - @copyright Copyright (c) 2022 Marco Ambrosini <marcoambrosini@pm.me>
+  -
+  - @author Marco Ambrosini <marcoambrosini@pm.me>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<docs>
+### Description
+
+General purpose text field component.
+Note: the inner html `input` element inherits all the attributes from the
+textField component so you can add things like `autocomplete`, `maxlength`
+and `minlength`.
+
+```
+<template>
+	<div class="wrapper">
+		<TextField :value.sync="text1"
+			label="Leading icon and trailing button"
+			trailing-button-icon="close"
+			:show-trailing-button="text1 !== ''"
+			@trailing-button-click="clearText">
+			<Magnify :size="16" />
+		</TextField>
+		<TextField :value.sync="text2"
+			label="Success state"
+			:success="true"
+			@trailing-button-click="clearText">
+		</TextField>
+		<TextField :value.sync="text3"
+			label="Error state"
+			:error="true"
+			@trailing-button-click="clearText">
+		</TextField>
+		<TextField :value.sync="text4"
+			label="Internal label"
+			placeholder="That can be used together with placeholder"
+			:label-visible="true"
+			trailing-button-icon="close"
+			:show-trailing-button="text4 !== ''"
+			@trailing-button-click="clearText">
+			<Lock :size="16" />
+		</TextField>
+		<div class="external-label">
+			<label for="$refs.textField.id">External label</label>
+			<TextField :value.sync="text5"
+				ref="textField"
+				:label-outside= "true"
```

Comment:
```suggestion
				:label-outside="true"
```
**Suggestion  (non-blocking)**